app/services/doc_auth/error_generator.rb

```diff
@@ -203,28 +101,29 @@ def get_image_metric_errors(processed_image_metrics)
 
       error_result
     end
+  end
 
-    def get_error_messages(liveness_enabled, response_info)
-      errors = Hash.new { |hash, key| hash[key] = Set.new }
-
-      if response_info[:doc_auth_result] != 'Passed'
-        response_info[:processed_alerts][:failed]&.each do |alert|
-          alert_msg_hash = ALERT_MESSAGES[alert[:name].to_sym]
+  class SelfieErrorHandler < ErrorHandler
+    include SelfieConcern
+    def handle(response_info)
+      liveness_enabled = response_info[:liveness_enabled]
+      get_selfie_error(liveness_enabled, response_info)
+    end
 
-          if alert_msg_hash.present?
-            field_type = alert[:side] || alert_msg_hash[:type]
-            errors[field_type.to_sym] << alert_msg_hash[:msg_key]
-          end
-        end
-      end
+    def is_selfie_failure(error)
+      error == Errors::SELFIE_FAILURE
```

Comment:
I think I may not be following, but does this mean `is_selfie_failure` would return `false` if `SELFIE_POOR_QUALITY` or `SELFIE_NOT_LIVE`?

Reply:
@night-jellyfish , renamed the function, basically this is a case that we present a general selfie error.